Doc/library/importlib.rst

```diff
@@ -775,6 +775,168 @@ ABC hierarchy::
         itself does not end in ``__init__``.
 
 
+:mod:`importlib.resources` -- Resources
+---------------------------------------
+
+.. module:: importlib.resources
+    :synopsis: Package resource reading, opening, and access
+
+**Source code:** :source:`Lib/importlib/resources.py`
+
+--------------
+
+.. versionadded:: 3.7
+
+This module leverages Python's import system to provide access to *resources*
+within *packages*.  If you can import a package, you can access resources
+within that package.  Resources can be opened or read, in either binary or
+text mode.
+
+Resources are roughly akin to files inside directories, though it's important
+to keep in mind that this is just a metaphor.  Resources and packages **do
+not** have to exist as physical files and directories on the file system.
+
+Loaders can support resources by implementing the :class:`ResourceReader`
+abstract base class.
+
+The following types are defined.
+
+.. class:: Package
```

Comment:
It looks misleading to describe it as a *class*, no?

Reply:
Any suggestions on what directive to use?

Reply:
I don't know. I think it's the first time a stdlib module exposes typing declarations. I'm not sure it's a good idea (see current python-dev discussion).

Reply:
Well, [typing](https://docs.python.org/3.7/library/typing.html) defines types as classes, so with the lack of a better alternative, I think it's fine.

Reply:
Yes... technically it's correct. What I mean is that a class is supposed to specify concrete behaviour, but being a type, `Package` doesn't have any methods or attributes of its own. So I don't see the point of mentioning it in the docs.

Reply:
It describes the expected interfaces for arguments in the API, so it seems valuable to me.  We'd still have to describe that *somewhere* and I don't want to have to repeat it for every method.

Reply:
Yeah... It might be worth discussing on python-dev to see what a recommended practice can look like.

Reply:
Ivan said that Guido prefers `data::` and that makes more sense in the face of PEP 560/563.  So I've made that switch in my branch for consistency.  I do still think it makes sense to document them because user code may want to use those types in their own annotations.